api/models.go

```diff
@@ -218,11 +220,31 @@ type Transaction struct {
 	CreatedAt       string      `json:"createdAt"`
 	SettledAt       *string     `json:"settledAt"`
 	AppId           *uint       `json:"appId"`
-	Metadata        Metadata    `json:"metadata,omitempty"`
+	Metadata        *Metadata   `json:"metadata,omitempty"`
 	Boostagram      *Boostagram `json:"boostagram,omitempty"`
 }
 
-type Metadata = map[string]interface{}
+type Metadata struct {
```

Comment:
Transaction metadata can be anything. It _could_ contain these fields. With this change it probably breaks other types of metadata

Reply:
No it will not break other types, we just ignore them while converting to the api metadata (which we use in transaction list). If you pass in other fields and do a `list_transactions` call, you'll see the other fields too from the database.

Reply:
But yeah having this here causes confusion as Metadata (and Boostagram) also exists in the db.Transaction type as `datatypes.JSON` but we again specify it differently here but not add it to `transactions.Transaction` struct (we only use these Metadata and Boostagram types in api > transactions.go)

Reply:
Done!